src/utility.js

```diff
@@ -25,7 +26,7 @@ export const combineReducers = reducers => {
           action
         )
         hasStateChanged =
-          hasStateChanged || nextStateForCurrentKey !== prevStateForCurrentKey
+          hasStateChanged || !isEqualObject(nextStateForCurrentKey, prevStateForCurrentKey)
```

Comment:
Why do we need deep compare? Both react and redux does shallow compare and if we deep compare on every render, it would be very expensive.

Reply:
Good point, both React and Redux use immutable state with shallow updates. I suppose deep compares are not needed since new objects are created on each state change and the comparison you have just checks the reference address of the object to see if it's changed. As long as people practice good state immutability all is good. Feel free to close this PR and related issue.

Thanks Ankita!

Reply:
Yes, please do not merge this... This library should work the same as Redux IMO

Reply:
I'm not sure what value your statement added here, but I did close this since she never did.

Reply:
I'm planning on bringing this library into a project.  With the PR and issue open I wasn't sure what the future intentions were.  I would use the project with this feature merged in so it was important.

Wasn't trying to "offend anyone", I just don't think this is correct behavior. Sorry

Reply:
You can use a dependency at a certain version w/o an additional feature. This was not to be merged, and was to be closed as inferred from the prior message. If you weren't for sure of intentions, asking always helps vs. demanding.